src/libraries/Microsoft.PowerFx.Core/Syntax/Nodes/TypeLiteralNode.cs

```diff
@@ -47,5 +51,148 @@ public override void Accept(TexlVisitor visitor)
 
         /// <inheritdoc />
         public override NodeKind Kind => NodeKind.TypeLiteral;
+
+        internal static bool IsValid(TypeLiteralNode typeNode, out List<TexlError> errors)
+        {
+            var validator = new Validator();
+            typeNode.TypeRoot.Accept(validator);
+            errors = validator.Errors;
+            return !errors.Any();
+        }
+
+        private class Validator : IdentityTexlVisitor
+        {
+            private readonly List<TexlError> _errors;
+
+            internal List<TexlError> Errors => _errors;
+
+            public Validator()
+            {
+                _errors = new List<TexlError>();
+            }
+
+            // Valid Nodes
+            public override void Visit(FirstNameNode node)
+            {
+            }
+
+            public override bool PreVisit(RecordNode node)
+            {
+                return true;
+            }
+
+            public override void PostVisit(RecordNode node)
```

Comment:
> public override void PostVisit(RecordNode node) [](http://example.com/codeflow?start=11&length=48)

It's no different than base.

Reply:
started without this but added override explicitly to ensure & show that we be doing anything in postvisit for these nodes

Reply:
If you are going to implement all methods here then you can just extend from TexlVisitor.